pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -521,3 +522,101 @@ func (this *SnapshotManager) CreateVolumeFromSnapshot(peID astrolabe.ProtectedEn
 	updatedID, err = astrolabe.NewProtectedEntityIDFromString(download.Status.VolumeID)
 	return
 }
+
+func (this *SnapshotManager) CreateVolumeFromSnapshotWithMetadata(peID astrolabe.ProtectedEntityID, metadata []byte, snapshotID string, backupRepositoryName string) (astrolabe.ProtectedEntityID, *v1api.Download, error) {
+	this.Infof("CreateVolumeFromSnapshotWithMetadata: Start creating restore for %s", peID.String())
+	// TODO(xyang): Will enable following code when astrolabe.Overwrite is ready
+	//config, err := rest.InClusterConfig()
+	//if err != nil {
+	//	this.WithError(err).Errorf("Failed to get k8s inClusterConfig")
+	//	return
+	//}
+	//pluginClient, err := plugin_clientset.NewForConfig(config)
+	//if err != nil {
+	//	this.WithError(err).Errorf("Failed to get k8s clientset with the given config: %v", config)
+	//	return
+	//}
+
+	veleroNs, exist := os.LookupEnv("VELERO_NAMESPACE")
+	if !exist {
+		this.Errorf("CreateVolumeFromSnapshot: Failed to lookup the env variable for velero namespace")
+		return astrolabe.ProtectedEntityID{}, nil, errors.New(fmt.Sprintf("failed to lookup the env variable for velero namespace"))
+	}
+
+	// Retrieve PVC PE TypeManager
+	peTM := this.Pem.GetProtectedEntityTypeManager("pvc")
```

Comment:
it is better to use peID.GetPeType() rather than hardcode "PVC"

Reply:
Done